Posting journal lets you hide all columns

lomamech opened the pull request:
Posting journal lets you hide all columns 
Prevent the user from hiding all posting journal columns

closes #999

lomamech committed:
Posting journal lets you hide all columns
Prevent the user from hiding all posting journal columns

jniles reviewed (changes requested):
I don't think this solution goes in the right direction.  The code is okay, but the user experience isn't good.

Here's my proposal:
 1. Use a `text-danger` notification in the modal whenever the users hides all columns.  If the user has at least one column visible, the notification should be hidden again.
 2. Disable the "submit" button if there are fewer than 1 visible columns.

Currently, this code broadcasts a change event on every click.  This is inefficient.  You could make this more efficient by moving the 

```js
this.gridApi.core.notifyDataChange(uiGridConstants.dataChange.COLUMN);      
```
call into the `submit()` method.

lomamech committed:
Review prevent hide all columns
- Using a text danger in notification in modal
- Prevent submit when there are zero visible column
- Check number of visible column on each Click

jniles commented on client/src/js/services/grid/GridColumns.js:
```diff
@@ -137,6 +137,36 @@ function GridColumnService(uiGridConstants, AppCache, Modal, util, $timeout) {
+   * This structure is used to count the number of selected columns
+   *
+   */
+  Columns.prototype.checkColumnMap = function checkColumnMap(columns) {
```
This is a well named and well described function.  Very nice!

jniles commented on client/src/partials/templates/modals/columnConfig.modal.js:
```diff
@@ -44,7 +47,14 @@ function ColumnsConfigModalController(ModalInstance, Columns) {
   }
 
+  function checkVisible(){
+    var columnNumber = Columns.checkColumnMap(vm.map);
```
In the future, a more meaningful name would be better - `Columns.checkColumnMap()` does not return a number - it return a boolean.  A better name is `hasEnoughColumns` or something along those lines.

jniles commented on client/src/partials/templates/modals/columnConfig.modal.js:
```diff
@@ -29,12 +29,15 @@ function ColumnsConfigModalController(ModalInstance, Columns) {
 
   // reset the column visibility to their default configuration
   function resetDefaults() {
+    vm.notification = false;
```
`vm.notification` could also use a nicer name, such as `vm.hasTooFewColumns`.

jniles reviewed (commented):
@lomamech, this code looks really good!

I cannot get this PR to work when I test it ... does it work on your machine?

lomamech committed:
Using the nice name for some variable
Recalculte the real default column valued true
Using alert alert-danger
